client/js/ace/mode-fbp.js

```diff
@@ -54,15 +54,14 @@ var ShHighlightRules = function() {
     var pointFloat = "(?:(?:" + intPart + "?" + fraction + ")|(?:" + intPart + "\\.))";
     var exponentFloat = "(?:(?:" + pointFloat + "|" +  intPart + ")" + ")";
     var floatNumber = "(?:" + exponentFloat + "|" + pointFloat + ")";
-    var fileDescriptor = "(?:&" + intPart + ")";
-    var fileDescriptor = "[a-zA-Z0-9_-]+\\/[a-zA-Z0-9_-]+";
+    var nodeTypes = "[(]([^:)]+)[:)]";
 
-    var variableName = "[a-zA-Z_][a-zA-Z0-9_]*";
-    var variable = "(?:(?:\\$" + variableName + ")|(?:" + variableName + "=))";
+    var nodeOptions = "[a-z_][a-z0-9_]*";
+    var variable = "(?:(?:\\$" + nodeOptions + ")|(?:" + nodeOptions + "=))";
 
     var builtinVariable = "(?:\\$(?:SHLVL|\\$|\\!|\\?))";
 
-    var func = "[A-Z0-9]+";
+    var ports = " [A-Z0-9\\[\\]]+ ";
```

Comment:
The leading spaces is a stop condition, otherwise it will color everything as port before and after the port definition.